### PR TITLE
Action Manager | Expose Checkable Action and Menu registration to Python

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Exposes multiple APIs from the Action and Menu managers to Python:
- RegisterCheckableAction
- UpdateAction
- RegisterMenu
- AddActionToMenu
- AddSeparatorToMenu
- AddSubMenuToMenu

Verified that a Python script can now add a checkable action to a menu and the update callback is called when the action is triggered.
```
import azlmbr.legacy.general as general
import azlmbr.action as action
import azlmbr.bus as bus
from PySide2.QtCore import Qt
from PySide2.QtWidgets import QDialog, QLabel, QVBoxLayout, QAction

def someCustomAction():
    panes = general.get_pane_class_names()
    test_pane = panes[6]
    
    if general.is_pane_visible(test_pane):
        general.close_pane(test_pane)
    else:
        general.open_pane(test_pane)

def someCustomCallBack():
    panes = general.get_pane_class_names()
    test_pane = panes[6]
    return general.is_pane_visible(test_pane)

properties = action.ActionProperties()
properties.name = "Custom Action"
properties.description = "A custom action defined via PySide"
properties.category = "Python"

action.ActionManagerPythonRequestBus(bus.Broadcast, "RegisterCheckableAction", "o3de.context.editor.mainwindow", "o3de.action.python.test", properties, someCustomAction, someCustomCallBack)

action.MenuManagerPythonRequestBus(bus.Broadcast, "AddActionToMenu", "o3de.menu.editor.file", "o3de.action.python.test", 42)
```
pythonCheckableActionAndMenuRegistration

In implementing these changes, I also took the chance to refactor a few classes to make them more consistent:
- renamed PythonEditorActionHandler to PythonActionManagerHandler to match the filename;
- changed the updateCallback from `AZStd::function<void(QAction*)>` to `AZStd::function<bool()>` - the function should return the checked state for the action. This allows to both reduce the access to the underlying action, and simplifies the process of reflecting this functionality to Python;
- Removed the redefinitions of the outcome type in the ActionManagerBus, everything now reuses the existing definition in AzToolsFramework;
- Moved the reflection of the Action Manager classes and buses to PythonActionManagerHandler instead of the PythonSystemComponent;
- Renamed the member variable and getter inside PythonEditorAction to reference the PyObject instead of a generic "handler" that would create some confusion;
- Similarly renamed the internal class PythonActionHandler to PythonFunctionObject.

I am working on introducing automated Python tests for this functionality, but the current status of the system would likely make them flakey (since the system is behind a flag, and there's no way to create an action context via Python yet) so the test will come in a follow-up PR. Everything that can be tested in unit tests is already covered by previous PRs.

### CODE REVIEW

File: Gems/EditorPythonBindings/Code/Source/ActionManager/PythonActionManagerHandler.h [hunk 1]
```diff
@@ -8,33 +8,55 @@
...
+            const AZStd::string& actionIdentifier,
+            const AzToolsFramework::ActionProperties& properties,
+            PythonEditorAction handler,
+            PythonEditorAction updateCallback) override;
+        AzToolsFramework::ActionManagerOperationResult TriggerAction(const AZStd::string& actionIdentifier) override;
+        AzToolsFramework::ActionManagerOperationResult UpdateAction(const AZStd::string& actionIdentifier) override;
+        
+        // MenuManagerPythonRequestBus overrides ...
+        AzToolsFramework::MenuManagerOperationResult RegisterMenu(const AZStd::string& identifier,
+            const AzToolsFramework::MenuProperties& properties) override;
```
Comment (hultonha) [inline Gems/EditorPythonBindings/Code/Source/ActionManager/PythonActionManagerHandler.h:53]: uber nit: Is it possible to move the first param to the next line to be consistent with the other function declarations?

File: Gems/EditorPythonBindings/Code/Source/ActionManager/MenuManagerBus.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    //! Bus to register menus and add menu items to them in the Editor via Python.
+    //! If writing C++ code, use the MenuManagerInterface instead.
+    class MenuManagerRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+
+        //! Register a new Menu to the Menu Manager.
+        virtual AzToolsFramework::MenuManagerOperationResult RegisterMenu(const AZStd::string& identifier,
+            const AzToolsFramework::MenuProperties& properties) = 0;
```
Comment (hultonha) [inline Gems/EditorPythonBindings/Code/Source/ActionManager/MenuManagerBus.h:29]: uber nit: Is it possible to move the first param to the next line to be consistent with the other function declarations?

**Review by hultonha — APPROVED** [review]
LGTM! I just have some super minor suggestions

File: Gems/EditorPythonBindings/Code/Source/ActionManager/PythonActionManagerHandler.h [hunk 3]
```diff
@@ -46,22 +68,24 @@ namespace EditorPythonBindings
...
         public:
-            explicit PythonActionHandler(PyObject* handler);
-            PythonActionHandler(const PythonActionHandler& obj);
-            PythonActionHandler(PythonActionHandler&& obj);
-            PythonActionHandler& operator=(const PythonActionHandler& obj);
-            virtual ~PythonActionHandler();
+            explicit PythonFunctionObject(PyObject* handler);
+            PythonFunctionObject(const PythonFunctionObject& obj);
+            PythonFunctionObject(PythonFunctionObject&& obj);
+            PythonFunctionObject& operator=(const PythonFunctionObject& obj);
```
[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply Gems/EditorPythonBindings/Code/Source/ActionManager/PythonActionManagerHandler.h:80]: Added explicit deletion

**Comment by AMZN-daimini** [comment]
AR failed cause I forgot to update the tests, it was on the back of my mind but I forgot it over the weekend :D Will fix and re-start it asap.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**